File: Qx/docs/trivia_difficulty.ts
/**
 * Enum defining Open Trivia Database (OTDB) difficulty levels.
 * For TypeScript frontend/backend use.
 */
export enum TriviaDifficulty {
  ANY = "any",
  EASY = "easy",
  MEDIUM = "medium",
  HARD = "hard"
}

/**
 * Mapping of TriviaDifficulty enum to human-readable names.
 */
export const TriviaDifficultyNames: Record<TriviaDifficulty, string> = {
  [TriviaDifficulty.ANY]: "Any Difficulty",
  [TriviaDifficulty.EASY]: "Easy",
  [TriviaDifficulty.MEDIUM]: "Medium",
  [TriviaDifficulty.HARD]: "Hard"
};

File: quiz_arena/lib/data/models/trivia_difficulty.dart
/// Enum định nghĩa mức độ khó của Open Trivia Database (OTDB).
/// Sử dụng cho ứng dụng di động Flutter.
enum TriviaDifficulty {
  any(value: 'any', displayName: 'Any Difficulty'),
  easy(value: 'easy', displayName: 'Easy'),
  medium(value: 'medium', displayName: 'Medium'),
  hard(value: 'hard', displayName: 'Hard');

  final String value;
  final String displayName;

  const TriviaDifficulty({required this.value, required this.displayName});

  /// Tìm kiếm độ khó từ giá trị API String nhận được.
  /// Trả về [TriviaDifficulty.any] nếu không tìm thấy.
  static TriviaDifficulty fromValue(String val) {
    return TriviaDifficulty.values.firstWhere(
      (e) => e.value == val,
      orElse: () => TriviaDifficulty.any,
    );
  }
}
